webgui/src/flipmouse/js/tabslot.js

```diff
@@ -1,13 +1,61 @@
 window.tabSlot = {};
 
+//require is only useful in electron.
+var inElectron = false;
+var activateOnProcess;
+
+//check every second, if a process is active, which should load a slot.
+setInterval(function() {
+	if (typeof activateOnProcess !== 'undefined') {
+		//load list of currently running processes
+		ps.lookup({}, function(err, resultList ) {
+			if (err) {
+				throw new Error( err );
+			}
+			//we just need process names
+			resultList = resultList.map(a => a.command);
+			//remove duplicates (we trigger on one process name anyway and the list is shorter for iter)
+			resultList = resultList.filter((v, i, a) => a.indexOf(v) === i);
+			//compare list of slots/process names against currently running ones
+			for (var i = 0; i < activateOnProcess.length; i++) {
```

Comment:
As far as I understand it this selects the corresponding slot, if any of the currently running applications has a slot mapping in the `activateOnProcess` list. But: (1) what if there are two processes running with an existing mapping? Then the selected slot will be random, (2) maybe it makes more sense to use the process/window with current focus to decide which slot to use?! then another library, like [this](https://github.com/octalmage/active-window) would be needed.

Reply:
Yes, good idea!